File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/comment/direct/Java/Go/s863488345_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)

	var mapNum int
	if scanner.Scan() {
		mapNum, _ = strconv.Atoi(scanner.Text())
	}

	for i := 0; i < mapNum; i++ {
		if i > 0 {
			fmt.Println()
		}

		var h, w int
		if scanner.Scan() {
			h, _ = strconv.Atoi(scanner.Text())
		}
		if scanner.Scan() {
			w, _ = strconv.Atoi(scanner.Text())
		}

		battleTown := make([][]byte, h)
		for row := 0; row < h; row++ {
			if scanner.Scan() {
				line := scanner.Text()
				battleTown[row] = []byte(line)
			}
		}

		var x, y int
		for row := 0; row < h; row++ {
			for col := 0; col < w; col++ {
				if battleTown[row][col] == '^' || battleTown[row][col] == 'v' || battleTown[row][col] == '<' || battleTown[row][col] == '>' {
					x, y = row, col
				}
			}
		}

		var bomberActNum int
		if scanner.Scan() {
			bomberActNum, _ = strconv.Atoi(scanner.Text())
		}
		bomberAct := ""
		if scanner.Scan() {
			bomberAct = scanner.Text()
		}

		for _, action := range bomberAct {
			switch action {
			case 'U':
				battleTown[x][y] = '^'
				if x-1 >= 0 && battleTown[x-1][y] == '.' {
					battleTown[x-1][y] = '^'
					battleTown[x][y] = '.'
					x--
				}
			case 'D':
				battleTown[x][y] = 'v'
				if x+1 < h && battleTown[x+1][y] == '.' {
					battleTown[x+1][y] = 'v'
					battleTown[x][y] = '.'
					x++
				}
			case 'L':
				battleTown[x][y] = '<'
				if y-1 >= 0 && battleTown[x][y-1] == '.' {
					battleTown[x][y-1] = '<'
					battleTown[x][y] == '.' // typo fix: should be '='
					y--
				}
			case 'R':
				battleTown[x][y] = '>'
				if y+1 < w && battleTown[x][y+1] == '.' {
					battleTown[x][y+1] = '>'
					battleTown[x][y] = '.'
					y++
				}
			case 'S':
				moveBomber(battleTown, x, y)
			}
		}

		for _, row := range battleTown {
			fmt.Println(string(row))
		}
	}
}

func moveBomber(battleTown [][]byte, a int, b int) {
	direction := battleTown[a][b]
	switch direction {
	case '^':
		currentRow := a - 1
		col := b
		for currentRow >= 0 {
			cell := battleTown[currentRow][col]
			if cell == '*' {
				battleTown[currentRow][col] = '.'
				return
			} else if cell == '#' {
				return
			} else {
				currentRow--
			}
		}
	case 'v':
		currentRow := a + 1
		col := b
		for currentRow < len(battleTown) {
			cell := battleTown[currentRow][col]
			if cell == '*' {
				battleTown[currentRow][col] = '.'
				return
			} else if cell == '#' {
				return
			} else {
				currentRow++
			}
		}
	case '<':
		currentCol := b - 1
		row := a
		for currentCol >= 0 {
			cell := battleTown[row][currentCol]
			if cell == '*' {
				battleTown[row][currentCol] = '.'
				return
			} else if cell == '#' {
				return
			} else {
				currentCol--
			}
		}
	case '>':
		currentCol := b + 1
		row := a
		for currentCol < len(battleTown[row]) {
			cell := battleTown[row][currentCol]
			if cell == '*' {
				battleTown[row][currentCol] = '.'
				return
			} else if cell == '#' {
				return
			} else {
				currentCol++
			}
		}
	}
} // End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/comment/direct/Java/Python/s863488345_commented.py
import sys

def move_bomber(a, b, battleTown, h, w):
    direction = battleTown[a][b]
    if direction == '^':
        current_x = a - 1
        current_y = b
        while current_x >= 0:
            cell = battleTown[current_x][current_y]
            if cell == '*':
                battleTown[current_x][current_y] = '.'
                break
            elif cell == '#':
                break
            else:
                current_x -= 1
    elif direction == 'v':
        current_x = a + 1
        current_y = b
        while current_x < h:
            cell = battleTown[current_x][current_y]
            if cell == '*':
                battleTown[current_x][current_y] = '.'
                break
            elif cell == '#':
                break
            else:
                current_x += 1
    elif direction == '<':
        current_x = a
        current_y = b - 1
        while current_y >= 0:
            cell = battleTown[current_x][current_y]
            if cell == '*':
                battleTown[current_x][current_y] = '.'
                break
            elif cell == '#':
                break
            else:
                current_y -= 1
    elif direction == '>':
        current_x = a
        current_y = b + 1
        while current_y < w:
            cell = battleTown[current_x][current_y]
            if cell == '*':
                battleTown[current_x][current_y] = '.'
                break
            elif cell == '#':
                break
            else:
                current_y += 1

def main():
    mapNum = int(sys.stdin.readline())
    for _ in range(mapNum):
        if _ > 0:
            print()
        h, w = map(int, sys.stdin.readline().split())
        battleTown = []
        current_x, current_y = -1, -1
        for j in range(h):
            line = sys.stdin.readline().strip()
            row = list(line)
            battleTown.append(row)
            for k in range(w):
                if row[k] in {'^', 'v', '<', '>'}:
                    current_x, current_y = j, k
        bomberActNum = int(sys.stdin.readline())
        bomberAct = sys.stdin.readline().strip()
        for cmd in bomberAct:
            if cmd == 'U':
                new_x = current_x - 1
                new_y = current_y
                if 0 <= new_x < h and 0 <= new_y < w and battleTown[new_x][new_y] == '.':
                    battleTown[current_x][current_y] = '.'
                    current_x = new_x
                    current_y = new_y
                    battleTown[new_x][new_y] = '^'
                else:
                    battleTown[current_x][current_y] = '^'
            elif cmd == 'D':
                new_x = current_x + 1
                new_y = current_y
                if 0 <= new_x < h and 0 <= new_y < w and battleTown[new_x][new_y] == '.':
                    battleTown[current_x][current_y] = '.'
                    current_x = new_x
                    current_y = new_y
                    battleTown[new_x][new_y] = 'v'
                else:
                    battleTown[current_x][current_y] = 'v'
            elif cmd == 'L':
                new_x = current_x
                new_y = current_y - 1
                if 0 <= new_x < h and 0 <= new_y < w and battleTown[new_x][new_y] == '.':
                    battleTown[current_x][current_y] = '.'
                    current_y = new_y
                    battleTown[new_x][new_y] = '<'
                else:
                    battleTown[current_x][current_y] = '<'
            elif cmd == 'R':
                new_x = current_x
                new_y = current_y + 1
                if 0 <= new_x < h and 0 <= new_y < w and battleTown[new_x][new_y] == '.':
                    battleTown[current_x][current_y] = '.'
                    current_y = new_y
                    battleTown[new_x][new_y] = '>'
                else:
                    battleTown[current_x][current_y] = '>'
            elif cmd == 'S':
                move_bomber(current_x, current_y, battleTown, h, w)
        for row in battleTown:
            print(''.join(row))
if __name__ == "__main__":
    main() # End of Code
